extension/solarwindsapmsettingsextension/extension.go

```diff
@@ -25,13 +32,63 @@ func newSolarwindsApmSettingsExtension(extensionCfg *Config, logger *zap.Logger)
 	return settingsExtension, nil
 }
 
-func (extension *solarwindsapmSettingsExtension) Start(_ context.Context, _ component.Host) error {
+func (extension *solarwindsapmSettingsExtension) Start(ctx context.Context, _ component.Host) error {
 	extension.logger.Debug("Starting up solarwinds apm settings extension")
-	_, extension.cancel = context.WithCancel(context.Background())
+	ctx, extension.cancel = context.WithCancel(ctx)
+	configOk := validateSolarwindsApmSettingsExtensionConfiguration(extension.config, extension.logger)
+	if configOk {
+		extension.conn, _ = grpc.Dial(extension.config.Endpoint, grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{})))
+		extension.logger.Info("Dailed to " + extension.config.Endpoint)
+		extension.client = collectorpb.NewTraceCollectorClient(extension.conn)
+		go func() {
+			ticker := newTicker(extension.config.Interval)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ticker.C:
+					refresh(extension)
+				case <-ctx.Done():
+					extension.logger.Debug("Received ctx.Done() from ticker")
```

Comment:
this is not typically how a component is stopped.

Reply:
I was referring to the [tailtracerReceiver example](https://opentelemetry.io/docs/collector/building/receiver/#implementing-the-receiver-component) to implement the extension. 
Can you help pointing me a typical example to stop a component?